pkg/backend/filestate/backend.go

```diff
@@ -128,7 +128,31 @@ func IsFileStateBackendURL(urlstr string) bool {
 
 const FilePathPrefix = "file://"
 
+// New constructs a new filestate backend using the default options
+// and the given project as the current project.
 func New(ctx context.Context, d diag.Sink, originalURL string, project *workspace.Project) (Backend, error) {
+	return NewWithOptions(ctx, d, originalURL, &Options{Project: project})
+}
+
+// Options defines the optional parameters for the filestate backend.
+type Options struct {
```

Comment:
I really don't like this Go pattern of struct options, you don't get any assistance from the compiler when adding things to ensure you think about each call site and whether it should/shouldn't be passing that option. 
I can understand this design for a public API where breaking changes need to be minimized but for our internal code where we know any use is confined to this repo and can just cascade out a fix I prefer the long list of parameters style, just so we can get some compiler assistance when adding/changing things.
But I think I'm in the minority here :(

Reply:
Repeating part of Slack chat here for visibility:

> I see and agree with the pain points.
> There are cases where I feel the readability gained by structs outweighs the pain/loss of safety and compiler enforced guarantees (one example: there are many parameters with the same type)—in those cases I'd argue you can use "find references" in your IDE or git grep, and fix anything that needs fixing—besides, the new fields should be optional anyway. [...] 
> The case that brought up this discussion isn't one where I would make [that argument], though. I'm happy to back out that change.